### ISSUE TITLE
Unable to use nested configuration

### ISSUE THREAD

**Issue by joshuaspence** [voxpupuli/puppet-telegraf#38]
I was trying to use the following Hiera configuration to create a Telegraf input:

```
telegraf::inputs:
  elasticsearch:
    cluster_health: true
    local: false
    servers: ['http://127.0.0.1']
    tags:
      role: 'some_role'
```

This resulted in the following (snippet) of configuration:

```
[[inputs.elasticsearch]]
  cluster_health = true
  local = false
  servers = ["http://127.0.0.1"]
  tags = {"role"=>"some_role"}
```

This is wrong and causes `telegraf` to fail:

```
Error parsing /etc/telegraf/telegraf.conf, toml: line 42: parse error
```

Rather, it should become;

```
[[inputs.elasticsearch]]
  cluster_health = true
  local = false
  servers = ["http://127.0.0.1"]
  [inputs.elasticsearch.tags]
    role = "some_role"
```

It looks like this is already supported by `telegraf::input`, so I think that the real issue here is that specifying inputs in `$telegraf::inputs` doesn't actually utilize the `telegraf::input` resource.

**Comment by doomnuggets** [comment]
Not sure if this is still an issue.
Solving this would require a slight change in your hiera parameterization.

```
telegraf::inputs:
  elasticsearch:
    cluster_health: true
    local: false
    servers: ['http://127.0.0.1']
    sections:
      'elasticsearch.tags':
        role: 'some_role'
```

Refer to the [README](README.md) Example 3

**Comment by yankcrime** [comment]
Fixed via #80.
